### PR TITLE
Info dump 10-13/64: Menus

### PR DESCRIPTION
**Pull request by End45**
Adds functions mostly related to menus, both dungeon mode menus and the main menu.

### CODE REVIEW

File: headers/types/common/enums.h [hunk 1]
```diff
@@ -2088,6 +2088,31 @@ enum item_id {
...
+    CATEGORY_POKE = 6, // Money
+    CATEGORY_UNK_7 = 7,
+    CATEGORY_OTHER = 8,
+    CATEGORY_ORBS = 9,
+    CATEGORY_LINK_BOX = 10,
+    CATEGORY_USED_TM = 11,
+    CATEGORY_CHESTS_1 = 12,
+    CATEGORY_CHESTS_2 = 13,
+    CATEGORY_CHESTS_3 = 14,
+    CATEGORY_EXCLUSIVE_ITEMS = 15
```
Comment (UsernameFodder) [inline headers/types/common/enums.h:2108]: I'm being picky, but let's put a trailing comma here 😛

**Review by UsernameFodder — CHANGES_REQUESTED** [review]
Mostly minor comments, looks good otherwise. I'll push up fixes for the unambiguous stuff shortly.

File: headers/types/common/enums.h [hunk 2]
```diff
@@ -2088,6 +2088,31 @@ enum item_id {
...
+    CATEGORY_TMS_HMS = 5,
+    CATEGORY_POKE = 6, // Money
+    CATEGORY_UNK_7 = 7,
+    CATEGORY_OTHER = 8,
+    CATEGORY_ORBS = 9,
+    CATEGORY_LINK_BOX = 10,
+    CATEGORY_USED_TM = 11,
+    CATEGORY_CHESTS_1 = 12,
+    CATEGORY_CHESTS_2 = 13,
+    CATEGORY_CHESTS_3 = 14,
```
[the remark this answers is not in the document]
Comment (End45) [reply headers/types/common/enums.h]: Sure

**Review by UsernameFodder — APPROVED** [review]
Thanks!

**Pull request merged**